xcell/tests/test_cls.py

```diff
@@ -203,3 +248,150 @@ def test_symmetric():
+    shutil.rmtree(tmpdir1)
+
+
+def test_cov_nonoverlap():
+    data = get_config(fsky=0.2, fsky2=0.2)
+    data['tracers']['Dummy__0']['dec0'] = 0.
+    data['tracers']['Dummy__1']['dec0'] = 180.
+    covc = Cov(data, 'Dummy__0', 'Dummy__0', 'Dummy__1', 'Dummy__1')
+    cov = covc.get_covariance()
+    shutil.rmtree(tmpdir1)
+    assert np.all(cov == 0)
+
+
+def test_cov_mmarg():
+    sm = 0.1
+    data = get_config(dtype='galaxy_shear')
+    data['tracers']['Dummy__0']['sigma_m'] = sm
+
+    # Homemade marginalized covariance
+    # First, get decoupled power spectra
+    # Theory power spectra
+    clf = ClFid(data, 'Dummy__0', 'Dummy__0')
+    _, cl = clf.get_ell_cl()
+    shutil.rmtree(tmpdir1)
+    # Binning
+    clc = Cl(data, 'Dummy__0', 'Dummy__0')
+    wp = clc.get_bandpower_windows()
+    shutil.rmtree(tmpdir1)
+    ncl, nbpw, _, nl = wp.shape
+    wp = wp.reshape((ncl*nbpw, ncl*nl))
+    cl = cl.reshape(ncl*nl)
+    cl = np.dot(wp, cl)
+    # Marginalized covariance term
+    covmargb = 4*sm**2*cl[:, None]*cl[None, :]
+
+    # Do with xCell
+    covc = Cov(data, 'Dummy__0', 'Dummy__0', 'Dummy__0', 'Dummy__0')
+    covmarg = covc.get_covariance_m_marg()
+    shutil.rmtree(tmpdir1)
+    assert np.amax(np.fabs(covmarg-covmargb))/np.mean(covmarg) < 1E-5
+
+
+@pytest.mark.parametrize('perm', [
+    [0, 0, 0, 1],  # 00, 02
+    [0, 0, 1, 1],  # 00, 22
+    [0, 1, 0, 0],  # 02, 00
+    [0, 1, 0, 1],  # 02, 02
+    [0, 1, 1, 1],  # 02, 22
+    [1, 1, 0, 0],  # 22, 00
+    [1, 1, 0, 1],  # 22, 02
+    [1, 1, 1, 1]])  # 22, 22
+def test_cov_spin0(perm):
+    nmp = [1, 2]
+    nmaps = [nmp[p] for p in perm]
+    ncls1 = nmaps[0]*nmaps[1]
+    ncls2 = nmaps[2]*nmaps[3]
+    data = get_config(fsky=1., fsky2=1.,
+                      dtype='galaxy_density',
+                      dtype2='galaxy_shear')
+    data['tracers']['Dummy__0']['noise_level'] = 1E-5
+    data['tracers']['Dummy__1']['noise_level'] = 1E-10
+    nbpw = len(data['bpw_edges'])
+
+    # Spin-2 covariance class
+    covc2 = Cov(data,
+                f'Dummy__{perm[0]}',
+                f'Dummy__{perm[1]}',
+                f'Dummy__{perm[2]}',
+                f'Dummy__{perm[3]}')
+    assert not covc2.spin0
+    cov2 = covc2.get_covariance()
+    assert cov2.shape == (ncls1*nbpw, ncls2*nbpw)
+    shutil.rmtree(tmpdir1)
+
+    # Spin-0 covariance class
+    data['cov']['spin0'] = True
+    covc0 = Cov(data,
+                f'Dummy__{perm[0]}',
+                f'Dummy__{perm[1]}',
+                f'Dummy__{perm[2]}',
+                f'Dummy__{perm[3]}')
+    assert covc0.spin0
+    cov0 = covc0.get_covariance()
+    assert cov0.shape == (ncls1*nbpw, ncls2*nbpw)
+    shutil.rmtree(tmpdir1)
+
+    if ncls1 == ncls2:
+        # Check that they are the same on all bandpowers
+        # except the first one
+        r = np.diag(cov2)/np.diag(cov0)-1
+        # This loops through EE, EB, BE, BB
+        for i in range(ncls1):
+            assert np.all(r[i::ncls1][1:] < 1E-5)
```

Comment:
Why is this `True`?

Reply:
in the full sky, the spin-0 and general approximations should be almost the same (except for the first bandpower).

Reply:
Oooh! I hadn't noticed that `fsky=1`, sorry!